mathics/builtin/numeric.py

```diff
@@ -824,7 +825,437 @@ def apply(self, n, base, evaluation, nr_elements=None):
+    
+    def apply_rational_with_base(self, n, b, evaluation):
+        'RealDigits[n_Rational, b_Integer]'
+        
+        expr = Expression('RealDigits', n)
+           
+        py_n = abs(n.value)
+        py_b = b.to_python()
+        
+        if check_finite_decimal(n.denominator().to_python()) == True and not py_b % 2 :
+            return self.apply_2(n, b, evaluation)
+        else:
+            exp = int(mpmath.ceil(mpmath.log(py_n, py_b)))
+            (head, tails) = convert_repeating_decimal(py_n.as_numer_denom()[0], py_n.as_numer_denom()[1], py_b)
+            
+            list_str = Expression('List')
+            for x in head:
+                if not x == '0':
+                    list_str.leaves.append(from_python(int(x)))
+            list_str.leaves.append(from_python(tails))
+            
+        return Expression('List', list_str, exp)
+    
+    def apply_rational_without_base(self, n, evaluation):
+        'RealDigits[n_Rational]'
+
+        return self.apply_rational_with_base(n, from_python(10), evaluation)
+ 
+    def apply(self, n, evaluation):
+        'RealDigits[n_]'
+        
+        # Handling the testcases that throw the error message and return the ouput that doesn't include `base` argument
+        if isinstance(n, Symbol):
+            return evaluation.message('RealDigits', 'ndig', n) if n.name.startswith('System`')  else Expression('RealDigits', n)
+        
+        if n.is_numeric():
+            return self.apply_2(n, from_python(10), evaluation)
+        
+    def apply_2(self, n, b, evaluation, nr_elements=None, pos=None):
+        'RealDigits[n_?NumericQ, b_Integer]'
+        
+        rational_no = True if isinstance(n, Rational) else False
+        expr = Expression('RealDigits', n)
+        py_b = b.to_python()
+        if isinstance(n, (Expression, Symbol, Rational)):
+            pos_len = abs(pos) + 1 if pos is not None and pos < 0 else 1
+            if nr_elements is not None:
+                n = Expression('N', n, int(mpmath.log(py_b ** (nr_elements + pos_len), 10)) + 1).evaluate(evaluation)
+            else:
+                if isinstance(n, Rational):
+                    n = Expression('N', n).evaluate(evaluation)
+                else:
+                    return evaluation.message('RealDigits', 'ndig', expr)
+        py_n = abs(n.value)
+        
+        if not py_b > 1:
+            return evaluation.message('RealDigits', 'rbase', py_b)
+                                      
+        if isinstance(py_n, complex):
+            return evaluation.message('RealDigits', 'realx', expr)
+        
+        if isinstance(n, Integer):
+            display_len = int(mpmath.floor(mpmath.log(py_n, py_b))) if py_n != 0 and py_n != 1 else int(1)
+        else:
+            display_len = int(Expression('N', Expression('Round', Expression('Divide', Expression('Precision', py_n), Expression('Log', 10, py_b)))).evaluate(evaluation).to_python())
```

Comment:
The evaluation returns an output with MachineReal type so we can't use get_int_value() here.